gulpfile.js

```diff
@@ -51,6 +52,82 @@ const { launch, connect } = require('hadouken-js-adapter');
 		console.log(`[${new Date().toLocaleTimeString()}] ${chalk[color][bgcolor](msg)}.`);
 	}
 
+	/** 
+	* Splits a string version with semantic versioning into an object with major, minor and patch versions
+	* Valid inputs are 'X.X.X' or 'vX.X.X'
+	*/
+	const createSemverObject = (version) => {
+		let tempVersionArray;
+		let semverObject;
+		if (!version || typeof version !== 'string') {
+			console.log(`Version must be type string`);
+			return;
+		}
+		// Split the version into a temp array.
+		if (version.startsWith('v')) {
+			tempVersionArray = version.split('v');
+			tempVersionArray = tempVersionArray[1].split('.');
+		} else {
+			tempVersionArray = version.split('.')
+		}
+		if (tempVersionArray)
```

Comment:
Counter to what one might think, an array in Javascript is always true.

For example
```
var hmm = ''.split(',');
console.log(Boolean(hmm));
```

prints `true`.

So I think you need to check the length of the array and also have an `else` for when the check fails:
```
		if (tempVersionArray.length != 3)
    {
		// Convert each array element to a number and store in the object.
		semverObject = {
			majorVersion: Number(tempVersionArray[0]) || null,
			minorVersion: Number(tempVersionArray[1]) || null,
			patchVersion: Number(tempVersionArray[2]) || null,
		};
    }
    else
    {
         return;
    }
		// If major, minor or patch versions are missing or not a number return nothing
		if (!semverObject.majorVersion || !semverObject.minorVersion || !semverObject.patchVersion) {
			return;
		}
		return semverObject;
	}
```

Reply:
Fixed that check. The error case is already handled because the function execution ends and will return automatically.